modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java

```diff
@@ -400,8 +401,21 @@ public ObjectRelationship updateObjectRelationship(
 			indexer.reindex(reverseObjectRelationship);
 		}
 
-		return _updateObjectRelationship(
+		ObjectRelationship newObjectRelationship = _updateObjectRelationship(
 			parameterObjectFieldId, deletionType, labelMap, objectRelationship);
+
+		if (StringUtil.equals(
```

Comment:
Keep the object relationship update on the return line and make this validation before that. The deletion type is already provided and the type won't change from the original instance.